src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2595,7 +2595,7 @@ protected function getAttributeValue($key)
 		// date fields without having to create a mutator for each property.
 		elseif (in_array($key, $this->getDates()))
 		{
-			if ($value) return $this->asDateTime($value);
+			if (!is_null($value)) return $this->asDateTime($value);
```

Comment:
@GrahamCampbell is the king of the cs.

But I think that this should be `if ( ! is_null($value))`.